src/main/java/bio/terra/common/iam/ProxiedAuthenticatedUserRequestFactory.java

```diff
@@ -43,10 +29,7 @@ public AuthenticatedUserRequest from(HttpServletRequest servletRequest) {
     } else {
       String authHeader = servletRequest.getHeader(AUTHORIZATION);
       if (authHeader != null) {
-        Matcher matcher = authorizationPattern.matcher(authHeader);
-        if (matcher.matches()) {
-          builder.setToken(matcher.group("token"));
-        }
+        BearerTokenParser.parseInto(authHeader, builder::setToken);
       }
     }
     return builder.build();
```

Comment:
Makes sense, would you recommend still having `build()` throw `IllegalStateExcpetion` per the [Java Practices Doc](https://engdoc.corp.google.com/eng/doc/devguide/java/practices/builders.md?cl=head#api), and try/catch/throw an auth specific exception here (aligned with the discussion about throwing in the bearer token parser utility method)?

Reply:
I think `build()` throwing an IllegalStateException still makes sense (the builder doesn't know it's getting bad values because the user isn't authenticated, it's just not getting what it needds). I think catching the IllegalStateException and rethrowing it as the cause of an auth specific exception would work fine. Or throwing an auth exception before passing any null values to the builder.